packages/api/lib/writeRecords/write-execution.js

```diff
@@ -131,6 +144,47 @@ const _writeExecutionRecord = ({
   return executionPgRecord;
 });
 
+/**
+ * Write execution record to databases and publish SNS message
+ *
+ * @param {Object} params
+ * @param {Object} params.dynamoRecord - Execution DynamoDB record to be written
+ * @param {Object} params.postgresRecord - Execution PostgreSQL record to be written
+ * @param {Object} params.knex - Knex client
+ * @param {Object} [params.executionModel] - DynamoDB execution model
+ * @param {Object} [params.executionPgModel] - PostgreSQL execution model
+ * @param {number} [params.updatedAt] - updatedAt timestamp to use when writing records
+ * @param {Object} [params.esClient] - Elasticsearch client
+ * @returns {Promise<Object>} - PostgreSQL execution record that was written to the database
+ */
+const _writeExecutionAndPublishSnsMessage = async ({
+  dynamoRecord,
+  postgresRecord,
+  knex,
+  executionModel,
+  executionPgModel,
+  updatedAt,
+  esClient,
+}) => {
+  const writeExecutionResponse = await _writeExecutionRecord(
+    {
+      dynamoRecord,
+      postgresRecord,
+      knex,
+      executionModel,
+      esClient,
+      executionPgModel,
+      updatedAt,
+    }
+  );
+  const translatedExecution = await translatePostgresExecutionToApiExecution(
```

Comment:
This is probably a dumb question we already discussed, but why are we recreating this translation given we already have the dynamo record, and the only functional addition is the createdAt/updatedAt values?  

Consider eliminating this translation, to eliminate needless postgres db load.     There's hidden debt/complexity in that change, so based on previous convo, I'm fine leaving it for a phase 3 reconsideration.

Reply:
Are we sure the timstamps are the only things that would have changed from the dynamo record? It's also possible that the `status` value would be unknown until after the record has been written, given the conditional logic. 

While I agree that the extra database lookups here aren't ideal, I feel like making partial updates to the dynamo record feels less safe and error-prone than doing the wholesale translation. 

Also, in phase 3, dynamo doesn't exist, so we can't really use that as the basis for what gets sent to the SNS topic.